### ISSUE TITLE
Performance regression with multiplication

### ISSUE THREAD

**Issue by dpsanders** [JuliaIntervals/IntervalArithmetic.jl#380]
There seems to be a performance regression on Julia 1.5:

```jl
julia> x = interval(1.3, 1.3)

julia> @btime $(Ref(x))[]*$(Ref(x))[]
  45.431 ns (6 allocations: 96 bytes)
```
This gives 
```
  13.907 ns (0 allocations: 0 bytes)
```
on 1.4.1

This regression only seems to affect the `:tight` mode of directed rounding, not e.g. the `:accurate` one, which has got faster.

cc @matsueushi

**Comment by dpsanders** [comment]
In particular, these basic operations should *never* allocate.

**Comment by dpsanders** [comment]
Actually also on 1.4. 

There is nothing wrong with the code per se; it's an issue with the Julia compiler not inlining, I guess because the functions are now quite long and complicated.

Fixed by #382.

**Comment by dpsanders** [comment]
We urgently need a good benchmark suite to pick up these kinds of performance regressions!

**Comment by lbenet** [comment]
I fully agree!

**Comment by dpsanders** [comment]
Fixed by #382
